js/common/modules/@arangodb/foxx/check-args.js

```diff
@@ -46,6 +48,39 @@ function normalizeMimeType (mime) {
   ]));
 }
 
+function isJoiSchema (value, expectOpaque) {
+  if (!value) return false;
+  if (value.isJoi) return true;
+  if (expectOpaque) return false;
+  if (typeof value !== "object") return false;
+  if (Array.isArray(value)) return false;
+  const values = Object.values(value);
+  if (!values.length) return false;
+  return values.every(value => isJoiSchema(value, true));
+}
+
+function createSchemaValidator (schema) {
+  if (schema.isJoi) {
+    return (value) => schema.validate(value);
```

Comment:
JOI provides its own `validate` method on its schema objects so we need to expose that instead of generating our own.